frontend/packages/sdk/src/pruntime/coders.ts

```diff
@@ -0,0 +1,131 @@
+import { U8aFixed } from '@polkadot/types'
+import { type AccountId } from '@polkadot/types/interfaces'
+import { BN, BN_ZERO, hexAddPrefix, hexToU8a, stringToHex, u8aToHex } from '@polkadot/util'
+import { sr25519Agree } from '@polkadot/wasm-crypto'
+import { phalaTypes } from '../options'
+import { encrypt } from '../utils/aes-256-gcm'
+import { randomHex } from '../utils/hex'
+import { generatePair } from './certificate'
+
+type ContractAddress = AccountId | string
+type LooseNumber = number | string | bigint | BN | null
+
+function encodeNumber(n?: LooseNumber, defaults = BN_ZERO) {
+  if (!n) {
+    return defaults
+  }
+  if (typeof n === 'bigint') {
+    return n
```

Comment:
Should we always use BN?

Reply:
The original Polkadot Contract API supports all of these types.: https://github.com/polkadot-js/api/blob/38209a3123a590c942e56a9f813c58b57be6ec62/packages/api-contract/src/types.ts#L75-L77
So use a helper function here.